java/ql/src/experimental/Security/CWE/CWE-1104/UsesInsecureDependency.ql

```diff
@@ -0,0 +1,23 @@
+/**
+ * @name Using Insecure Dependency 
+ * @description A dependency imported into your project is either deprecated or will cease to be supported in the near future.
+ * @kind problem
+ * @problem.severity error
+ * @precision high
+ * @id java/microsoft/using-insecure-dependency
+ * @tags security
+ *       external/cwe/cwe-1104
+ */
+import java
+
+
+predicate usesInsecureDependency(Import i) {
+  i.toString().splitAt(" ", 1) in [ "com.microsoft.aad.adal4j.*", 
+    							    "AuthenticationContext",
```

Comment:
Why is this looking for particular class names? Can we just look for any import matching "com.microsoft.aad.adal4j.%"?